### PR TITLE
Reduce required method of Implementation

### PR DESCRIPTION
**Pull request by cheyang**
### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/controllers/runtime_controller.go [hunk 1]
```diff
@@ -91,7 +91,10 @@ func (r *RuntimeReconciler) ReconcileInternal(ctx cruntime.ReconcileRequestConte
 			ctx.Log.Info("the dataset can't be bound to the runtime, because it's already bound to another runtime ",
 				"dataset", dataset.Name)
 			dataset = nil
+			err = fmt.Errorf("the dataset can't be bound to the runtime, because it's already bound to another runtime %s", dataset.Name)
```
Comment (TrafalgarZZZ) [inline pkg/controllers/runtime_controller.go:94]: `dataset.Name` will lead to NPE here for the reason that `dataset` just be set to `nil`

Comment (cheyang) [reply pkg/controllers/runtime_controller.go:94]: Thanks for spotting it!

**Review by TrafalgarZZZ — APPROVED** [review]
/lgtm

**Review by xieydd — APPROVED** [review]
/lgtm

**Pull request merged**